src/main/kotlin/com/example/gymi/global/security/jwt/TokenProvider.kt

```diff
@@ -0,0 +1,117 @@
+package com.example.gymi.global.security.jwt
+
+import com.example.gymi.domain.auth.exception.RoleNotExistException
+import com.example.gymi.domain.user.enums.Role
+import com.example.gymi.global.security.auth.AuthDetailService
+import com.example.gymi.global.security.exception.TokenExpiredException
+import com.example.gymi.global.security.exception.TokenInvalidException
+import com.example.gymi.global.security.jwt.properties.JwtProperties
+import com.example.gymi.global.security.jwt.properties.TokenTimeProperties
+import io.jsonwebtoken.Claims
+import io.jsonwebtoken.ExpiredJwtException
+import io.jsonwebtoken.Jwts
+import io.jsonwebtoken.SignatureAlgorithm
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
+import org.springframework.security.core.Authentication
+import org.springframework.stereotype.Component
+import java.security.Key
+import java.time.ZonedDateTime
+import java.util.*
+import javax.servlet.http.HttpServletRequest
+
+@Component
+class TokenProvider(
+
+        private val jwtProperties: JwtProperties,
+        private val tokenTimeProperties: TokenTimeProperties,
+        private val authDetailService: AuthDetailService,
+) {
+    companion object {
+
+        const val ACCESS_TYPE = "access"
+        const val REFRESH_TYPE = "refresh"
+        const val TOKEN_PREFIX = "Bearer "
+        const val AUTHORITY = "authority"
+    }
+
+    val accessExpiredTime: ZonedDateTime
+        get() = ZonedDateTime.now().plusSeconds(tokenTimeProperties.accessTime)
+
+    val refreshExpiredTime: ZonedDateTime
+        get() = ZonedDateTime.now().plusSeconds(tokenTimeProperties.refreshTime)
+
+    fun generateAccessToken(email: String, role: Role): String =
+            generateToken(email, ACCESS_TYPE, jwtProperties.accessSecret, tokenTimeProperties.accessTime, role)
+
+    fun generateRefreshToken(email: String, role: Role): String =
+            generateToken(email, REFRESH_TYPE, jwtProperties.refreshSecret, tokenTimeProperties.refreshTime, role)
+
+    fun resolveToken(req: HttpServletRequest): String? {
+
+        val token = req.getHeader("Authorization") ?: return null
+        return parseToken(token)
+    }
+
+    fun exactEmailFromRefreshToken(refresh: String): String {
+
+        return getTokenSubject(refresh, jwtProperties.refreshSecret)
+    }
+
+    fun exactRoleFromRefreshToken(refresh: String): Role {
+
+        return when (getTokenBody(refresh, jwtProperties.refreshSecret).get(AUTHORITY, String::class.java)) {
```

Comment:
```suggestion
        val authority = getTokenBody(refresh, jwtProperties.refreshSecret)
            .get(AUTHORITY, String::class.java)
        return when (authority) {
```
이런식으로 변수로 빼는게 가독성이 더 좋을거같아요.

Reply:
수정했습니다!